src/libasr/asr_verify.cpp

```diff
@@ -440,6 +440,20 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         verify_unique_dependencies(x.m_dependencies, x.n_dependencies,
                                    x.m_name, x.base.base.loc);
 
+        // Dependencies of the function should be from same symbol table.
```

Comment:
@certik Do you think this logic is on the right track for checking the that dependencies & function should be from same symbol table ?